### PR TITLE
Sendgrid API v3

### PR DESCRIPTION
**Pull request by roelio**
Updated to use the new sendgrid API

Still needs some work with respect to the sendgrid specific headers that are used in the package.
Also some testing would be nice and the docs need some updates ;-)

### CODE REVIEW

File: sgbackend/mail.py [hunk 1]
```diff
@@ -51,61 +44,80 @@ def send_messages(self, emails):
...
-                count += 1
-            except sendgrid.SendGridClientError:
-                if not self.fail_silently:
-                    raise
-            except sendgrid.SendGridServerError:
-                if not self.fail_silently:
-                    raise
+            mail = self._build_sg_mail(email)
+            response = self.sg.client.mail.send.post(request_body=mail)
+            count += 1
```
Comment (elbuo8) [inline sgbackend/mail.py]: We shouldn't crash applications if `sendgrid` is down of there is another type of exception raised. 

Mind adding this `try/catch`?

File: sgbackend/mail.py [hunk 2]
```diff
@@ -51,61 +44,80 @@ def send_messages(self, emails):
...
 
-            if "Filters" in email.extra_headers:
-                mail.smtpapi.data['filters'] = email.extra_headers["Filters"]
+                # if "Filters" in email.extra_headers:
+                #    mail.smtpapi.data['filters'] = email.extra_headers["Filters"]
 
-            for attachment in email.attachments:
-                mail.add_attachment_stream(attachment[0], attachment[1])
+                # for attachment in email.attachments:
+                #    mail.add_attachment_stream(attachment[0], attachment[1])
```
Comment (elbuo8) [inline sgbackend/mail.py]: Delete the historical comments 👍

File: sgbackend/mail.py [hunk 3]
```diff
@@ -51,61 +44,80 @@ def send_messages(self, emails):
...
+                mail.add_attachment(attach)
 
-        if email.extra_headers:
-            if "Reply-To" in email.extra_headers:
-                reply_to = rfc822.parseaddr(email.extra_headers["Reply-To"])[1]
-                mail.set_replyto(reply_to)
+                # if email.extra_headers:
+                # if "Reply-To" in email.extra_headers:
+                # reply_to = rfc822.parseaddr(email.extra_headers["Reply-To"])[1]
+                # personalization.set_replyto(reply_to)
```
Comment (elbuo8) [inline sgbackend/mail.py]: We should def support this.

**Comment by andriisoldatenko** [comment]
any updates?

**Comment by andriisoldatenko** [comment]
I think we can close this PR because of https://github.com/elbuo8/sendgrid-django/pull/26

**Pull request closed without merging**